sklearn/utils/extmath.py

```diff
@@ -360,6 +356,164 @@ def randomized_svd(M, n_components, n_oversamples=10, n_iter='auto',
         return U[:, :n_components], s[:n_components], V[:n_components, :]
 
 
+def _normalize_power_iteration(x, power_iteration_normalizer):
+    """Normalize the matrix when doing power iterations for stability."""
+    if power_iteration_normalizer == "none":
+        return x
+    elif power_iteration_normalizer == "LU":
+        Q, _ = linalg.lu(x, permute_l=True)
+        return Q
+    elif power_iteration_normalizer == "QR":
+        Q, _ = linalg.qr(x, mode="economic")
+        return Q
+    else:
+        raise ValueError("Unrecognized normalization method `%s`" %
+                         power_iteration_normalizer)
+
+
+def randomized_pca(A, n_components, n_oversamples=10, n_iter="auto",
+                   power_iteration_normalizer="auto", flip_sign=True,
+                   random_state=0):
+    """Computes a truncated randomized PCA decomposition.
```

Comment:
> PCA is not really a decomposition so I wouldn't call PCA decomposition.

Isn't it? I always thought of PCA as a decomposition of variance. The matrix is "decomposed" into loadings and the directions of highest variance.

Would PCA projection be better?

> renaming `randomized_pca` into `randomized_pca_sparse`

Some renaming might fine, but it might be worth pointing out that the only real difference between `randomized_svd` and `randomized_pca` is that `randomized_pca` computes the SVD of the centered `x` matrix, without explicitly needing to center it. So basically, if you pre-centered `x`, the results would be exactly the same from  `randomized_svd` as from `randomized_pca`. And since centering dense matrices is cheap, you'd only ever do this on sparse matrices.

Reply:
>PCA is not really a decomposition

What I meant is that it's not a decomposition (or factorization) like SVD is: It does not decomposes your matrix into a product of other matrices. It just computes a projection matrix.

I 100% agree with you about the difference between randomized_svd and randomized_pca, hence my suggestion about the naming, and the docstring explaining all this.